api/v1alpha1/lua_types.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+// Lua defines a Lua extension
+// Only one of Source or Disabled must be set
+//
+// +kubebuilder:validation:XValidation:rule="has(self.source) ? (!has(self.disabled) || self.disabled == false) : (has(self.disabled) && self.disabled == true)",message="Exactly one of source or disabled must be set."
+type Lua struct {
+	// Source is the LuaSource that will be executed for this extension.
+	//
+	// +optional
+	Source *LuaSource `json:"source"`
+	// Disabled is the status of the extension.
+	// Used for disabling Lua extensions at a route level.
+	//
+	// +optional
+	Disabled *bool `json:"disabled,omitempty"`
```

Comment:
Do we need a `Disabled` knob here?

Reply:
@zhaohuabing for usecases where Lua is defined at GW level but we want to exclude some routes?

Reply:
>Besides, the EEP at the HTTPRoute level will override the one at the Gateway level.

Yes, but there should be a lever to turn it off altogether, rather than just override. WDYT?

Reply:
>I think this is a common issue for all the xPolicies, so prefer to solve it on the global level if needed

Removed disable for now, we can solve this later if needed once the bulk of Lua validation and implementation is done.